O365/sharepoint.py

```diff
@@ -236,8 +240,41 @@ def get_list_columns(self):
 
         return [self.list_column_constructor(parent=self, **{self._cloud_data_key: column})
                 for column in data.get('value', [])]
-
-
+    
+    def update_list_item(self, item_id, update):
+        """Updates a list item with a given update
+           
+        :param item_id: List item id to update
+        :param update: Data to update list item with {'col_name': 'Updated Data'}
+        
+        :rtype: SharepointListItem
+        """
+        
+        url = self.build_url(self._endpoints.get('get_item_by_id').format(item_id=item_id) + '/fields')
+        
+        response = self.con.patch(url, update)
+        
+        data = response.json()
+        
+        return self.get_item_by_id(item_id)
```

Comment:
This does another api call.

You can avoid this by updating the inner properties that the user has updated.

Update method calls should return True/False flagging the success.

The changes are implicit when the call succeeds.

Reply:
Removed the calling of the item from the API. 

Since the item does not have to be declared, and updates can be made with just the ID number, would it be more appropriate to treat the list like the mailbox, and the items like the message?  

This would move the update to the item class and the changes would be tracked within each item?  Was not sure what level the user should be able to update the list.  

Or am I missing something that could easily be done at the list level, like collecting the item ID, and fields to change and then pass that to the API in an update.

Reply:
Yep that's exactly what I was saying on the main review comment.
The update should happen in the item itself, and the item should be tracking the it's internal state to flag where there's been a change.

Then a simple item.update() (without any arguments) should "commit" only the properties changed to the servers.